subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/fixture/GradleProfilerBuildExperimentRunner.java

```diff
@@ -95,6 +100,10 @@ public void run(BuildExperimentSpec experiment, MeasuredOperationList results) {
                     results.add(measuredOperation);
                 }
             });
+            if (jfrProfileTargetDir != null) {
```

Comment:
You already set the generator to `NOOP` if the target dir is `null` so there shouldn't be a need for this check.

It seems to me that `flameGraphGenerator` should know about the JFR target dir instead of keeping it here. Same for `Profiler`, which could be created in the constructor, too.

Reply:
Both things are now instantiated in the constructor. As soon as all of the flame graph handling lives in Gradle profiler itself, we don't need to do anything on the Gradle side anymore.

Reply:
:+1: